suite-native/message-system/src/messageSystemMiddleware.ts

```diff
@@ -0,0 +1,32 @@
+import { isAnyOf } from '@reduxjs/toolkit';
+
+import { createMiddleware } from '@suite-common/redux-utils';
+import {
+    messageSystemActions,
+    categorizeMessages,
+    getValidMessages,
+} from '@suite-common/message-system';
+import { selectAccountsSymbols } from '@suite-common/wallet-core';
+
+const isAnyOfMessageSystemAffectingActions = isAnyOf(messageSystemActions.fetchSuccessUpdate);
+
+export const messageSystemMiddleware = createMiddleware((action, { next, dispatch, getState }) => {
+    if (isAnyOfMessageSystemAffectingActions(action)) {
+        const { config } = action.payload;
+
+        const enabledNetworks = selectAccountsSymbols(getState());
+
+        const validMessages = getValidMessages(config, {
```

Comment:
Create some note somewhere that this has to be expanded when device and transport (bluetooth, cable) added.

Reply:
noted somewhere (in my head) 😄 

good idea, I just don't know where to put it now, I believe we will keep that in mind 👍